inspirehep/modules/workflows/tasks/submission.py

```diff
@@ -330,21 +332,28 @@ def filter_keywords(obj, eng):
 @with_debug_logging
 def prepare_keywords(obj, eng):
     """Prepares the keywords in the correct format to be sent"""
-    prediction = obj.extra_data.get('keywords_prediction', {})
-    if not prediction:
-        return
-
+    prediction_keywords = get_value(obj.extra_data, 'keywords_prediction.keywords', [])
     keywords = obj.data.get('keywords', [])
-    for keyword in prediction.get('keywords', []):
-        # TODO: differentiate between curated and guessed keywords
-        keywords.append(
-            {
-                'value': keyword['label'],
-                'source': 'curator' if keyword.get('curated') else 'magpie',
-            }
-        )
-
-    obj.data['keywords'] = keywords
+    for keyword in prediction_keywords:
+        if keyword.get('curated'):
+            keywords.append(
+                {
+                    'value': keyword['label'],
+                    'source': 'curator',
+                }
+            )
```

Comment:
no, we don't need this thing (related to magpie too), you can remove it.

Reply:
:+1: